client/src/components/Home/components/Banners/Banners.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { offerImages } from "../../../../utility/groupImage";
+import styles from "./Banners.module.scss";
+import useInterval from "@src/customHooks/useInterval";
+
+const Banners: any = () => {
+  const [data, setData] = useState([]);
+  const [transX, setTransX] = useState(0);
+
+  useInterval(() => {
+    if (-(data.length - 1) * 100 === transX) setTransX(0);
+    else setTransX(transX - 100);
+  }, 3000);
+
+  useEffect(() => {
+    axios.get("http://localhost:3000/banners").then((response: any) => {
+      setData(response.data);
```

Comment:
extract the api logic to some external file. Component should not dependent on axios or any other such logic